apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -43,13 +44,19 @@ export class ProjectBuildCache {
   private readonly _project: RushConfigurationProject;
   private readonly _localBuildCacheProvider: FileSystemBuildCacheProvider;
   private readonly _cloudBuildCacheProvider: CloudBuildCacheProviderBase | undefined;
+  private readonly _buildCacheEnabled: boolean;
   private readonly _projectOutputFolderNames: string[];
   private readonly _cacheId: string | undefined;
 
+  public get buildCacheEnabled(): boolean {
```

Comment:
How does this setting relate to this logic:

https://github.com/microsoft/rushstack/blob/ae6d9138fa61ffa82a3045918882c9dcd9261460/apps/rush-lib/src/logic/taskRunner/ProjectBuilder.ts#L366-L374

🤔 Thinking about this, `BUILD_CACHE_ENABLED=false` can override `buildCacheEnabled=true` from the config file.  But if we have a config file like this:
```js
{
  "buildCacheEnabled": false,
  "cacheProvider": "local-only"
}
```
then I'm not sure `BUILD_CACHE_ENABLED=true` would successfully reverse that.  (Is there a realistic scenario where we would want that to work?)

Reply:
Yes, I definitely want that to work!  And it did work locally, but that was before your most recent fix to this logic in master.

I pulled on that thread a bit more and I think what makes the most sense is just move the new env var check all the way up to `BuildCacheConfiguration`.  This fixes not only that scenario, but also allows `rush update-cloud-credentials` and `rush write-build-cache` actions to work "as if" `buildCacheEnabled` was `true`.